docs/index.rst

```diff
@@ -1,12 +1,101 @@
-.. currentmodule:: celery_batches
+##############
+celery-batches
+##############
 
-.. automodule:: celery_batches
+Experimental task class that buffers messages and processes them as a list. Task
+requests are buffered in memory (on a worker) until either the flush count or
+flush interval is reached. Once the requests are flushed, they are sent to the
+task as a list of :class:`~celery_batches.SimpleRequest` instances.
 
-    **API**
+It is possible to return a result for each task request by calling
+``mark_as_done`` on your results backend. Returning a value from the Batch task
+call is only used to provide values to signals and does not populate into the
+results backend.
 
-    .. autoclass:: Batches
-        :members:
-        :undoc-members:
-    .. autoclass:: SimpleRequest
-        :members:
-        :undoc-members:
+.. warning::
+
+    For this to work you have to set
+    :setting:`worker_prefetch_multiplier` to zero, or some value where
+    the final multiplied value is higher than ``flush_every``. Note that Celery
+    will attempt to continually pull data into memory if this is set to zero.
+    This can cause excessive resource consumption on both Celery workers and the
+    broker when used with a deep queue.
+
+    In the future we hope to add the ability to direct batching tasks
+    to a channel with different QoS requirements than the task channel.
+
+**Simple Example**
+
+A click counter that flushes the buffer every 100 messages, and every
+10 seconds.  Does not do anything with the data, but can easily be modified
+to store it in a database.
+
+.. code-block:: python
+
+    # Flush after 100 messages, or 10 seconds.
+    @app.task(base=Batches, flush_every=100, flush_interval=10)
+    def count_click(requests):
+        from collections import Counter
+        count = Counter(request.kwargs['url'] for request in requests)
+        for url, count in count.items():
+            print('>>> Clicks: {0} -> {1}'.format(url, count))
+
+
+Then you can ask for a click to be counted by doing::
+
+    >>> count_click.delay('http://example.com')
+
+**Example returning results**
+
+An interface to the Web of Trust API that flushes the buffer every 100
+messages, and every 10 seconds.
+
+.. code-block:: python
+
+    import json
+    import urllib.parse
+    import urllib.request
+    import urlparse
```

Comment:
Might be best to leave the changes from `requests` to a separate PR?

Reply:
Rebased out